### PR TITLE
Image download links

### PR DESCRIPTION
**Pull request by ham-k7dev**
I've made a few changes to my meshchat:
- Octet-stream for attachments, inline and content-type for images.
- Relative download link for attachments.
- Open in new tab (affects images, attachments continue to work normally).

(I access my AREDN node in an odd way sometimes and using an absolute link instead of a relative link makes assumption about DNS resolution when I'm using the IP.)

### CODE REVIEW

**Comment by hickey** [comment]
Thank you for you contribution. 

I like the direction you are going here and have a few comments and thoughts.

File: meshchat [hunk 1]
```diff
@@ -538,8 +538,22 @@ function file_download()
...
+    if ext == "jpg" or ext == "jpeg" then
+        content_type = "image/jpeg"
+    elseif ext == "png" then
+        content_type = "image/png"
+    elseif ext == "gif" then
+        content_type = "image/gif"
+    else
+        print("Content-Disposition: attachment; filename=\"" .. file .. "\";\r")
+        content_type = "application/octet-stream"
+    end
```
Comment (hickey) [inline meshchat:554]: While this is a good first start, I am thinking that it might be more beneficial to start to introduce full MIME type support. Take a look at https://github.com/lunarmodules/lua-mimetypes. By using `lua-mimetypes` a lot more file formats can be supported and shorten this to just a couple of lines.

File: www/files.js [hunk 2]
```diff
@@ -110,7 +110,7 @@ function load_files() {
                     port = ':8080'
                 }
             }
-            html += '<td><a href="http://' + aredn_domain(data.files[i].node) + port + '/cgi-bin/meshchat?action=file_download&file=' + encodeURIComponent(data.files[i].file) + '">' + data.files[i].file + '</a></td>';
```
Comment (hickey) [inline www/files.js:113]: The `aredn_domain()` part is actually pretty important. The file that you are referencing may not be on the server that you are on. Was there an issue that you were trying to resolve?

**Review by hickey — CHANGES_REQUESTED** [review]

File: meshchat (hunk 1 above)
Comment (ham-k7dev) [reply meshchat:554]: I would counter with: "application/octet-stream" does handle all filetypes correctly, PDFs, ZIPs, EXEs, etc. (Because it downloads it and at that point, it's up to your computer to decide what to do with it.) And even if lua-mimetypes were included, you'd still need to conditionally allow through images as non-attachments, if you wanted them in a browser window and not downloaded. Also, in the small memory environment, this is a few lines of code, but lua-mimtypes is 300 or so more lines and doesn't cover the ~2,000 lines in a type Linux system. (/etc/mime.types)

I am happy to do it either way though.

File: www/files.js (hunk 2 above)
Comment (ham-k7dev) [reply www/files.js:113]: "The file that you are referencing may not be on the server that you are on."
I'm curious how/when that happens.

I'm accessing my AREDN node through the WAN IP currently. (Internally from my own network.) That's causing some weirdness unless I decide to manually plug into LAN. I admit this is a "my configuration" issue.

Comment (hickey) [reply www/files.js:113]: Let's say you have a 3 MeshChat instances (A, B and C) running on the same service name so that they are synchronized. A file gets uploaded to A. At the next sync period B and C will learn of this file and add it to their file list. B and C do not receive a copy of the file but just a reference to it. When users on B or C look at their file list they will see that the file is available. Upon selecting the file the link they receive pulls the file from A. 

There is not file syncing in MeshChat currently. There is an issue (#7) to discuss and explore file syncing options. 

The problem you are having is that AREDN nodes are not designed or purposed for access through their WAN interface. AREDN nodes and devices on the AREDN network resolve each other in the local.mesh domain which will not resolve on the WAN interface.

Comment (ham-k7dev) [reply www/files.js:113]: "There is not file syncing in MeshChat currently." <--- Aha. Now it makes sense, thanks! :+1:

File: meshchat (hunk 1 above)
Comment (ham-k7dev) [reply meshchat]: I've added more image mime types from lua-mimetypes. I'm not sure what it would add to include other mime types when `application/octet-stream` has been handling those fine.
